daffodil-test/src/test/scala/org/apache/daffodil/runtime1/layers/AISPayloadArmoringLayer.scala

```diff
@@ -183,17 +113,25 @@ class AISPayloadArmoringOutputStream(jos: java.io.OutputStream) extends OutputSt
 
   override def close(): Unit = {
     if (!closed) {
-      val ba = baos.toByteArray()
-      val dis = InputSourceDataInputStream(ba)
-      val finfo = new FormatInfoForAISDecode()
-      val cb = CharBuffer.allocate(256)
-      while ({ val numDecoded = dec.decode(dis, finfo, cb); numDecoded > 0 }) {
-        cb.flip()
-        IOUtils.write(cb, jos, iso8859)
-        cb.clear()
+      val ba = baos.toByteArray
+      using(InputSourceDataInputStream(ba)) { dis =>
+        val finfo = FormatInfoForAISDecode
+        val cb = CharBuffer.allocate(256)
+        //
+        // TODO: This is not a supportable API. We want to reuse the bitsCharset features of daffodil
+        //  for this non-byte-sized charset decoding. But this finfo object (a trait on the ParseOrUnparseState)
+        //  should not be part of the API
```

Comment:
Are you talking only about the AIS Payload Armoring Layer API, or about the layer API in general?  I haven't noticed the finfo object used elsewhere in this PR, just in this function.

Reply:
I will clarify. I'm talking about how this AIS layer is using the BitsCharset which is an internal thing (now anyway) so while the layer could be built in to Daffodil, it can't be written by an outsider without depending on Daffodil internals.